addons/binding/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/SpotifySession.java

```diff
@@ -0,0 +1,1299 @@
+     */
+    private void refreshToken() {
+        final String authString = Base64.getEncoder()
+                .encodeToString(String.format("%s:%s", clientId, clientSecret).getBytes());
+        logger.debug("Sending Spotify Web API token refresh request");
+
+        String content = "grant_type=refresh_token&refresh_token=" + refreshToken;
+        String contentType = "application/x-www-form-urlencoded";
+
+        for (int i = 0; i < HTTP_CLIENT_RETRY_COUNT; i++) {
+            logger.debug("Spotify Web API call attempt: {}", i + 1);
+            try {
+                String url = "https://accounts.spotify.com/api/token";
+                ContentResponse response = httpClient.POST(url).header("Authorization", "Basic " + authString)
+                        .content(new StringContentProvider(content), contentType)
+                        .timeout(HTTP_CLIENT_TIMEOUT, TimeUnit.SECONDS).send();
+
+                logger.debug("Response Code: {}", response.getStatus());
+                logger.trace("Response Data: {}", response.getContentAsString());
+
+                if (response.getStatus() == 200) {
+                    Gson gson = new Gson();
+                    SpotifyWebAPIRefreshResult test = gson.fromJson(response.getContentAsString(),
+                            SpotifyWebAPIRefreshResult.class);
+                    accessToken = test.accessToken;
+                    tokenValidity = test.getExpiresIn();
+
+                    spotifyPlayer.setChannelValue(SpotifyBindingConstants.CHANNEL_ACCESSTOKEN,
+                            new StringType(accessToken));
+
+                    return;
+                } else if (response.getStatus() == 400) {
+                    logger.error("Response: {} - verify that Spotify Client ID and Client Secret are correct!",
+                            response.getContentAsString());
+
+                }
+            } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                logger.debug("Exception caught while calling Spotify Web API for token refresh: ", e);
+            }
+            logger.debug("Attempt {} failed.", i + 1);
+        }
+        logger.error("Giving up on accessing Spotify Web API. Check network connectivity!");
+    }
+
+    @Override
+    public void run() {
+        try {
+            refreshToken();
+        } catch (Exception ex) {
+            logger.error("Exception caught in the run() of the scheduled token refresh.", ex);
+            throw (ex);
+        }
+    }
+
+    private String requestReply(String method, String url, String requestData) {
+
+        WebAPICallTask task = new WebAPICallTask(method, url, requestData);
+        Future<String> future = syncExecutorService.submit(task);
+
+        try {
+            while (!future.isDone()) {
+                Thread.sleep(ASYNC_RESULT_WAIT_TIME);
+            }
+            return future.get();
+
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception